src/distdl/nn/conv_feature_new.py

```diff
@@ -0,0 +1,388 @@
+
+    This class provides the user interface to a distributed convolutional
+    layer, where the input (and output) tensors are partitioned in
+    feature-space only.
+
+    The base unit of work is given by the input/output tensor partition.  This
+    class requires the following of the tensor partitions:
+
+    1. :math:`P_x` over input tensor :math:`x` has shape :math:`1 \times
+       P_{\text{c_in}} \times 1 \times \dots \times 1`.
+
+    The output partition, :math:`P_y`, is assumed to be the same as the
+    input partition.
+
+    The first dimension of the input/output partitions is the batch
+    dimension,the second is the channel dimension, and remaining dimensions
+    are feature dimensions.
+
+    The learnable weight and bias terms does not have their own partition.
+    They is stored at the 0th rank (index :math:`(0, 0,\dots, 0)` of
+    :math:`P_x`.  Each worker in :math:`P_x` does have their own local
+    convolutional layer but only one worker has learnable coefficients.
+
+    All inputs to this base class are passed through to the underlying PyTorch
+    convolutional layer.
+
+    Parameters
+    ----------
+    P_x :
+        Partition of input tensor.
+    in_channels :
+        (int)
+        Number of channels in the input image
+    out_channels :
+        (int)
+        Number of channels produced by the convolution
+    kernel_size :
+        (int or tuple)
+        Size of the convolving kernel.
+    stride :
+        (int or tuple, optional)
+        Stride of the convolution. Default: 1
+    padding :
+        (int or tuple, optional)
+        Zero-padding added to both sides of the input. Default: 0
+    padding_mode :
+        (int or tuple, optional)
+        'zeros', 'reflect', 'replicate' or 'circular'. Default: 'zeros'
+        .. warning::
+            Currently, only padding_mode = 'zeros' is supported.
+    dilation :
+        (int or tuple, optional)
+        Spacing between kernel elements. Default: 1
+    groups :
+        (int, optional)
+        Number of blocked connections from input channels to output channels. Default: 1
+    bias :
+        (bool, optional)
+        If True, adds a learnable bias to the output. Default: True
+    buffer_manager :
+        (BufferManager, optional)
+        Optional DistDL buffer manager. Default: None
+    """
+
+    # Convolution class for base unit of work.
+    TorchConvType = None
+
+    def __init__(self,
+                 P_x,
+                 in_channels,
+                 out_channels,
+                 kernel_size,
+                 stride=1,
+                 padding=0,
+                 padding_mode='zeros',
+                 dilation=1,
+                 groups=1,
+                 bias=True,
+                 buffer_manager=None):
+
+        super(DistributedFeatureConvBase, self).__init__()
+
+        # P_x is 1 x 1 x P_d-1 x ... x P_0
+        self.P_x = P_x
+
+        if padding_mode != 'zeros':
```

Comment:
If we are using `F.pad`, why is `'zeros'` the only legal mode?

Reply:
There is a bit of logic to ensure the padding mode is set properly, as the padding mode variable differs from PyTorch ConvNd to F.pad. I'll be pushing changes that adds this functionality.